### ISSUE TITLE
Extra call to `p_lin()` inside of `p_2h()` which already accepts linear Pk as one of its argument

### ISSUE THREAD

**Issue by MinhMPA** [cmbant/CAMB#133]
Inside the halofit.f90 module, there is the function
```
FUNCTION p_2h(this,k,plin,lut,cosm)
```
that computes the two-halo term of the nonlinear Pk.

This function accepts the linear Pk `plin` as one of its argument. However, inside `p_2h()`, for the default model, which is Mead et al. 2020 i.e. `imead=3`, instead of using this `plin`, `p_2h()` actually executes a call to
```
REAL FUNCTION p_dewiggle(k, z, sigv, cosm)
```
which in turn does an extra call to
```
FUNCTION p_lin(k,z,itype,cosm)
```

I guess this probably could be avoided and potentially lead to unintentionally error, e.g. when modifying the code. Or maybe I'm missing something.
I could make a pull request, if necessary.

**Comment by cmbant** [comment]
Look like one for @alexander-mead

**Comment by alexander-mead** [comment]
Yeah, I think you could replace the `lut%z` argument with `plin` on line `1708` and then remove the call to `p_lin` on line `1782` without affecting the running of the code in any way. Please feel free to change this. The tests should fail if this breaks anything.

**Comment by cmbant** [comment]
Try https://github.com/cmbant/CAMB/tree/haloclean

**Comment by alexander-mead** [comment]
Looks good to me!
